src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -327,6 +327,49 @@ class RichSparkContext(val sc: SparkContext) extends AnyVal {
       .map(l => Line(l, None, file))
 }
 
+class HeadRDD[T](prev: RDD[T], num: Int, exact: Boolean)(implicit tct: ClassTag[T]) extends RDD[T](prev) {
+
+  private var nLastPart = 0
+  private var lastPart = 0
+
+  def getPartitions: Array[Partition] = {
+    val buf = new mutable.ArrayBuffer[Partition]
+
+    var nvp = prev.sparkContext.runJob(prev,(it: Iterator[T]) => it.size, Seq(0))
+    if(exact) {
+      while (nvp.sum < num && nvp.size < prev.partitions.length) {
+        nvp = nvp ++ prev.sparkContext.runJob(prev, (it: Iterator[T]) => it.size,
+          Range(nvp.length,
+            nvp.length +
+              math.min(((num - nvp.sum) / (nvp.sum / nvp.length).toDouble).ceil.toInt, prev.partitions.length)))
+      }
+
+      def f(n: Int, i: Int): Unit = {
+        if (n + nvp(i) > num) {
+          lastPart = i
+          nLastPart = num - n
+        } else {
+          f(n + nvp(i), i + 1)
+        }
+      }
+      f(0, 0)
+    }else{
+      lastPart = math.min((num / nvp.sum.toDouble).ceil.toInt, prev.partitions.length)
```

Comment:
If you have ints-in ints-out, I think the following idiom is more standard: `(num + nvp.sum - 1) / nvp.sum`.